api/auth.proto

```diff
@@ -0,0 +1,43 @@
+syntax = "proto3";
+
+package envoy.api.v2;
+
+message AuthConfig {
+
+  // Check peer identity using X.509 certificate
+  message X509 {
+    enum VerifyType {
+      option allow_alias = true;
+      DEFAULT = 0;
+      ALL = 0;
+      ANY = 1;
+    }
+
+    // should all/any condition(s) matches
+    VerifyType verify_type = 1;
+
+    // NOTE: another level of indirect to get has_XXX behaviour
```

Comment:
Do we need this? Why can't we just use empty list?

Reply:
the verify list acts as a white-list, with an **empty** list allow none while a **nil** list allow all (don't do the check).

i will add more comment about this.

Reply:
after second thouhg, i made nil/empty list behavir the same - match none.